### DISCUSSION TITLE
List of working devices?

### DISCUSSION THREAD

**Discussion by weebee42** [wmbusmeters/wmbusmeters#926] [Q&A]
Hi all,

first of all my compliments and respect to the authors and contributors for this great solution. I did my first experiments with wmbusmeters, a terratec RTL-SDR USB-dongle and 2 ultrasonic heatmeters. And straight away I achieved an full integration with HASS.

I'm wondering if the list with working devices in the wmbusmeters/README.md is maintained/updatet or if there is such reference somewhere else. If so, maybe I can add another heat meter: **Zelsius 5 IUF** from **Zenner**.

I was successful with the c5isf driver. The only functional difference is, as far as I understand, that some telegrams named T1A1 and T1A2 in the driver are omitted: 
- prev_month_date and total_energy_prev_month_kwh are not delivered for the last 14 but only for 1 month,
- total_volume_prev_month_m3 is not delivered alltogether. 
Here is sample from the log:
`telegram=|5E44496A7200921088047A720050052F2F_0406150100000413F71E0000840406C500000082046CE121043B00000000042D0000000002590517025D2D0E02FD17000084800106EE0000008280016CE123948001AE25000000002F2F2F2F2F2F|+333`
...
`(shell) env "METER_JSON={"media":"heat","meter":"c5isf","name":"xxxx","id":"xxxxxxxx","total_energy_consumption_kwh":277,"total_volume_m3":7.927,"status":"OK","prev_1_month":"2023-03-01","prev_1_month_kwh":238,"due_energy_consumption_kwh":197,"due_date":"2023-01-01","volume_flow_m3h":0,"power_kw":0,"total_energy_consumption_last_month_kwh":238,"max_power_last_month_kw":0,"flow_temperature_c":58.93,"return_temperature_c":36.29,"timestamp":"2023-03-23T23:08:22Z","device":"rtlwmbus[cmd_0]","rssi_dbm":149}"`

But concerning the known devices I'm still a bit confused:
I guess, the driver name points towards Zelsius 5 ISF (a model quite similar to mine, just mechanical). But as far as I could find, is neither the driver or Z(C)elsius nor Zenner or Engelmann Sensostar (the matching json from the mqtt_discovery part in the ha_addon) mentioned anywhere.

So may be I'm sending owls to Athens, if so please excuse me. This is also my very first contribution in Github.
Please let me know, if I can provide any more related useful information.

Thanks a lot!

**Comment by weetmuts** [comment]
So the (w)mbus protocol has alot of standardized fields that can be automatically translated into total_kwh, total_m3 etc.
Therefore there is a lot of overlap between drivers, the standard fields will be ok. 
However for historical data, the meter telegram often just tags a total_m3 with storage=1 or storage 15 or something else.
What value is stored in storage 1, is it one month ago? last year, etc. You do not know until you read
the manual for the meter or compare the values with the values on the display.

The same is true for the status and error bits. 

In this case your telegram has the exact mfct/type/version combo that triggers the c5isf driver which indicates
it is the same meter hardware+software. Often the branding change on the meters. Sometimes the first driver gets
the name of that meter brand, then the meter contents gets resold under other brands, so it might look a
bit odd, but the driver is working perfectly.

Your telegram decodes all values. I.e. they are green.

https://wmbusmeters.org/analyze/5E44496A7200921088047A720050052F2F0406150100000413F71E0000840406C500000082046CE121043B00000000042D0000000002590517025D2D0E02FD17000084800106EE0000008280016CE123948001AE25000000002F2F2F2F2F2F
